bitnami/spark/templates/statefulset-worker.yaml

```diff
@@ -0,0 +1,192 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "spark.fullname" . }}-worker
+  labels:
+    app.kubernetes.io/name: {{ include "spark.name" . }}
+    helm.sh/chart: {{ include "spark.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/component: worker
+spec:
+  serviceName: {{ template "spark.fullname" . }}-headless
+  replicas: {{ .Values.worker.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "spark.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/component: worker
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "spark.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        helm.sh/chart: {{ include "spark.chart" . }}
+        app.kubernetes.io/component: worker
+    {{- if .Values.worker.nodeSelector }}
+      nodeSelector:
+      {{- toYaml .Values.worker.nodeSelector | nindent 8 }}
+    {{- end }}
+    {{- with .Values.worker.affinity }}
+      affinity:
+      {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.worker.tolerations }}
+      tolerations:
+      {{- toYaml . | nindent 8 }}
+    {{- end }}
+    spec:
+      {{- include "spark.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.worker.securityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.worker.securityContext.fsGroup }}
+        runAsUser: {{ .Values.worker.securityContext.runAsUser }}
+      {{- end }}
+      containers:
+        - name: spark-worker
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.worker.webPort }}
+              protocol: TCP
+          {{- if .Values.worker.clusterPort }}
+            - name: cluster
+              containerPort: {{ .Values.worker.clusterPort }}
+          {{- end }}
+          volumeMounts:
+          {{- if .Values.worker.configurationConfigMap }}
+          - name: config
+            mountPath: '/opt/bitnami/spark/conf/'
+          {{- end }}
+          {{- if .Values.security.certificatesSecret }}
+            - name: certs
+              mountPath: '/opt/bitnami/spark/conf/certs'
+              readOnly: true
+          {{- end }}
+          env:
+          - name: SPARK_MODE
+            value: "worker"
+          {{- if .Values.worker.debug }}
+          - name: BASH_DEBUG
+            value: "1"
+          - name: NAMI_DEBUG
+            value: "1"
+          - name: NAMI_LOG_LEVEL
+            value: "trace8"
+          {{- end }}
+          - name: SPARK_DAEMON_MEMORY
+            value: {{ .Values.worker.daemonMemoryLimit | quote }}
+          ## There are some environment variables wich existence needs
```

Comment:
`wich` -> `which`...or maybe you wanted to use `whose`?

The comment is not very clear to me